test/ttmlir/Conversion/TTIRToLinalg/ttir.mlir

```diff
@@ -0,0 +1,64 @@
+    return %1 : tensor<32x32xf32>
+  }
+
+  func.func @add_with_broadcast(
+    %arg0: tensor<32x32xf32>,
+    %arg1: tensor<32x1xf32>,
+    %arg2: tensor<32x32xf32>
+  ) -> tensor<32x32xf32> {
+    %1 = "ttir.add"(%arg0, %arg1, %arg2) <{operandSegmentSizes = array<i32: 2, 1>}> : (tensor<32x32xf32>, tensor<32x1xf32>, tensor<32x32xf32>) -> tensor<32x32xf32>
+    // CHECK: %{{.+}} = tensor.collapse_shape
+    // CHECK: %{{.+}} = tensor.empty()
+    // CHECK: %{{.+}} = linalg.broadcast ins(%{{.+}} : tensor<{{.*xf32}}>) outs(%{{.+}} : tensor<{{.*xf32}}>) dimensions
+    // CHECK: %{{.+}} = linalg.add ins(%{{.+}}, %{{.+}} : tensor<{{.*xf32}}>, tensor<{{.*xf32}}>) outs(%{{.+}} : tensor<{{.*xf32}}>) -> tensor<{{.*xf32}}>
+
+    return %1 : tensor<32x32xf32>
+  }
+
+  func.func @add_with_broadcast_1(
+    %arg0: tensor<32x1xf32>,
+    %arg1: tensor<32x32x32xf32>,
+    %arg2: tensor<32x32x32xf32>
+  ) -> tensor<32x32x32xf32> {
+    %1 = "ttir.add"(%arg0, %arg1, %arg2) <{operandSegmentSizes = array<i32: 2, 1>}> : (tensor<32x1xf32>, tensor<32x32x32xf32>, tensor<32x32x32xf32>) -> tensor<32x32x32xf32>
+    // CHECK: %{{.+}} = tensor.collapse_shape
+    // CHECK: %{{.+}} = tensor.empty()
+    // CHECK: %{{.+}} = linalg.broadcast ins(%{{.+}} : tensor<{{.*xf32}}>) outs(%{{.+}} : tensor<{{.*xf32}}>) dimensions =
+    // CHECK: %{{.+}} = linalg.add ins(%{{.+}}, %{{.+}} : tensor<{{.*xf32}}>, tensor<{{.*xf32}}>) outs(%{{.+}} : tensor<{{.*xf32}}>) -> tensor<{{.*xf32}}>
+    return %1 : tensor<32x32x32xf32>
+  }
+
+  func.func @add_with_broadcast_2(
+    %arg0: tensor<32x1x32xf32>,
+    %arg1: tensor<32x1x1xf32>,
+    %arg2: tensor<32x1x32xf32>
+  ) -> tensor<32x1x32xf32> {
+    %1 = "ttir.add"(%arg0, %arg1, %arg2) <{operandSegmentSizes = array<i32: 2, 1>}> : (tensor<32x1x32xf32>, tensor<32x1x1xf32>, tensor<32x1x32xf32>) -> tensor<32x1x32xf32>
+    // CHECK: %{{.+}} = tensor.collapse_shape
+    // CHECK: %{{.+}} = tensor.empty()
+    // CHECK: %{{.+}} = linalg.broadcast ins(%{{.+}} : tensor<{{.*xf32}}>) outs(%{{.+}} : tensor<{{.*xf32}}>) dimensions
+    // CHECK: %{{.+}} = linalg.add ins(%{{.+}}, %{{.+}} : tensor<{{.*xf32}}>, tensor<{{.*xf32}}>) outs(%{{.+}} : tensor<{{.*xf32}}>) -> tensor<{{.*xf32}}>
+    return %1 : tensor<32x1x32xf32>
+  }
+
+  func.func @add_with_broadcast_3(
+    %arg0: tensor<32xf32>,
+    %arg1: tensor<32x32xf32>,
+    %arg2: tensor<32x32xf32>
+  ) -> tensor<32x32xf32> {
+    %1 = "ttir.add"(%arg0, %arg1, %arg2) <{operandSegmentSizes = array<i32: 2, 1>}> : (tensor<32xf32>, tensor<32x32xf32>, tensor<32x32xf32>) -> tensor<32x32xf32>
+    // CHECK: %{{.+}} = tensor.empty()
+    // CHECK: %{{.+}} = linalg.broadcast ins(%{{.+}} : tensor<{{.*xf32}}>) outs(%{{.+}} : tensor<{{.*xf32}}>) dimensions
+    // CHECK: %{{.+}} = linalg.add ins(%{{.+}}, %{{.+}} : tensor<{{.*xf32}}>, tensor<{{.*xf32}}>) outs(%{{.+}} : tensor<{{.*xf32}}>) -> tensor<{{.*xf32}}>
+    return %1 : tensor<32x32xf32>
+  }
```

Comment:
Simiarly, I try to test variety of broadcast + collapse cases here, but let me know if you have suggestions for other cases.  (Should I add negative cases where broadcasting fails? etc)

Reply:
Negative tests is positive here, but if you add them you won't test your code since we are already testing for broadcastable operands using traits.